File: app/src/main/java/com/esbati/keivan/persiancalendar/Utils/PreferencesHelper.java
package com.esbati.keivan.persiancalendar.Utils;

import android.content.SharedPreferences;
import android.content.pm.PackageInfo;
import android.preference.PreferenceManager;
import android.widget.Toast;

import com.esbati.keivan.persiancalendar.Controllers.ApplicationController;

/**
 * Created by Esbati on 2/13/2016.
 */
public class PreferencesHelper {

    //Setting Toggles Keys
    public final static String KEY_ANIMATION_SELECTION = "showSelectionAnimation";
    public final static String KEY_NOTIFICATION_SHOW = "showNotification";
    public final static String KEY_NOTIFICATION_ACTIONS = "showNotificationAction";
    public final static String KEY_NOTIFICATION_PRIORITY = "notificationPriority";


    public static boolean isOptionActive(String key, boolean defaultValue){
        SharedPreferences settings = PreferenceManager
                .getDefaultSharedPreferences(ApplicationController.getContext());

        return settings.getBoolean(key, defaultValue);
    }

    public static void setOption(String key, boolean isActive){
        SharedPreferences settings = PreferenceManager
                .getDefaultSharedPreferences(ApplicationController.getContext());
        SharedPreferences.Editor editor = settings.edit();
        editor.putBoolean(key, isActive);
        editor.commit();
    }

    public static boolean toggleOption(String key, boolean defaultValue){
        SharedPreferences settings = PreferenceManager
                .getDefaultSharedPreferences(ApplicationController.getContext());
        SharedPreferences.Editor editor = settings.edit();

        boolean value = settings.getBoolean(key, defaultValue);
        editor.putBoolean(key, !value);
        editor.commit();

        return !value;
    }

    public static String loadString(String key, String defaultValue){
        SharedPreferences settings = PreferenceManager
                .getDefaultSharedPreferences(ApplicationController.getContext());
        String value = settings.getString(key, defaultValue);

        return value;
    }

    public static void saveString(String key, String value){
        SharedPreferences settings = PreferenceManager
                .getDefaultSharedPreferences(ApplicationController.getContext());
        SharedPreferences.Editor editor = settings.edit();
        editor.putString(key, value);
        editor.commit();
    }

    public static int loadInt(String key, int defaultValue){
        SharedPreferences settings = PreferenceManager
                .getDefaultSharedPreferences(ApplicationController.getContext());
        int intValue = settings.getInt(key, defaultValue);

        return intValue;
    }

    public static void saveInt(String key, int intValue){
        SharedPreferences settings = PreferenceManager
                .getDefaultSharedPreferences(ApplicationController.getContext());
        SharedPreferences.Editor editor = settings.edit();
        editor.putInt(key, intValue);
        editor.commit();
    }
}

File: app/src/main/java/com/esbati/keivan/persiancalendar/repository/PreferencesHelper.kt
package com.esbati.keivan.persiancalendar.repository

import android.content.SharedPreferences
import android.preference.PreferenceManager
import androidx.core.app.NotificationCompat
import com.esbati.keivan.persiancalendar.components.ApplicationController

/**
 * Created by Shubham on 4/15/2019.
 */
object PreferencesHelper {

    //region Setting Toggles Keys
    const val KEY_ANIMATION_SELECTION = "showSelectionAnimation"
    const val KEY_NOTIFICATION_SHOW = "showNotification"
    const val KEY_NOTIFICATION_ACTIONS = "showNotificationAction"
    const val KEY_NOTIFICATION_PRIORITY = "notificationPriority"
    const val KEY_PLAY_NOTE = "playNote"
    //endregion

    private val preferences: SharedPreferences =
            PreferenceManager.getDefaultSharedPreferences(ApplicationController.getContext())

    private fun save(operation: (SharedPreferences.Editor) -> Unit) {
        preferences.edit()
                .also(operation::invoke)
                .apply()
    }

    //region Specific operations
    var isAnimationSelectionActive: Boolean
        get() = preferences.getBoolean(KEY_ANIMATION_SELECTION, false)
        set(value) = save { it.putBoolean(KEY_ANIMATION_SELECTION, value) }

    var shouldShowNotification: Boolean
        get() = preferences.getBoolean(KEY_NOTIFICATION_SHOW, true)
        set(value) = save { it.putBoolean(KEY_NOTIFICATION_SHOW, value) }

    var isNotificationActionsActive: Boolean
        get() = preferences.getBoolean(KEY_NOTIFICATION_ACTIONS, true)
        set(value) = save { it.putBoolean(KEY_NOTIFICATION_ACTIONS, value) }

    var notificationPriority: Int
        get() = preferences.getInt(KEY_NOTIFICATION_PRIORITY, NotificationCompat.PRIORITY_MAX)
        set(value) = save { it.putInt(KEY_NOTIFICATION_PRIORITY, value) }

    var playNote : Boolean
        get() = preferences.getBoolean(KEY_PLAY_NOTE, true)
        set(value) = save { it.putBoolean(KEY_PLAY_NOTE, value) }

    //endregion

    //region Public methods
    fun isOptionActive(key: String, defaultValue: Boolean = false) = preferences.getBoolean(key, defaultValue)

    fun setOption(key: String, isActive: Boolean) = save { it.putBoolean(key, isActive) }

    fun toggleOption(key: String, defaultValue: Boolean = true): Boolean {
        return preferences.getBoolean(key, defaultValue).not()
                .also { setOption(key, it) }
    }

    fun loadString(key: String, defaultValue: String? = null) =
            preferences.getString(key, null) ?: defaultValue

    fun saveString(key: String, value: String) = save { it.putString(key, value) }

    fun loadInt(key: String, defaultValue: Int = 0) = preferences.getInt(key, defaultValue)

    fun saveInt(key: String, value: Int) = save { it.putInt(key, value) }

    //endregion
}